registrar.go

```diff
@@ -93,7 +93,7 @@ func (r Registry) GetDriverInterfaces() []interface{} {
 func (r Registry) FilterForCompatible(ctx context.Context) Drivers {
 	var wg sync.WaitGroup
 	mutex := &sync.Mutex{}
-	state := make(map[int]*Driver)
+	state := make([]*Driver, len(r.Drivers))
 
 	for index, elem := range r.Drivers {
 		index := index
```

Comment:
maybe we need a nil check here instead?

Reply:
true, a nil check here would also make sense ( although this is not the issue the  [1addda6](https://github.com/jacobweinstock/registrar/pull/1/commits/1addda69fb8485e3b96fabc9644811642e41cfc1) tries to address)